src/commands/android/subcommands/connect/emulator.ts

```diff
@@ -0,0 +1,65 @@
+import colors from 'ansi-colors';
+import inquirer from 'inquirer';
+
+import Logger from '../../../../logger';
+import {launchAVD} from '../../adb';
+import {Options, Platform} from '../../interfaces';
+import {getBinaryLocation} from '../../utils/common';
+import {execBinarySync} from '../../utils/sdk';
+import {showMissingBinaryHelp} from '../common';
+
+export async function connectAVD(options: Options, sdkRoot: string, platform: Platform): Promise<boolean> {
+  try {
+    const avdmanagerLocation = getBinaryLocation(sdkRoot, platform, 'avdmanager', true);
+    if (avdmanagerLocation === '') {
+      showMissingBinaryHelp('avdmanager');
+
+      return false;
+    }
+
+    const installedAvds = execBinarySync(avdmanagerLocation, 'avdmanager', platform, 'list avd -c');
+    if (installedAvds === null) {
+      Logger.log(`${colors.red('\nFailed to fetch installed AVDs.')} Please try again.\n`);
+
+      return false;
+    } else if (installedAvds === '') {
+      Logger.log(`${colors.yellow('No installed AVD found.')}\n`);
+      Logger.log('To see the list of installed AVDs, run the following command:');
+      Logger.log(colors.cyan('  npx @nightwatch/mobile-helper android list --avd\n'));
+
+      return false;
+    }
+
+    const installedAVDList = installedAvds.split('\n').filter(avd => avd !== '');
+
+    if (options.avd && !installedAVDList.includes(options.avd as string)) {
+      Logger.log(colors.yellow('Provided AVD not found!\n'));
+      options.avd = '';
+    }
+
+    if (!options.avd) {
+      const avdAnswer = await inquirer.prompt({
+        type: 'list',
+        name: 'avdName',
+        message: 'Select the AVD to connect:',
+        choices: installedAVDList
+      });
+      options.avd = avdAnswer.avdName;
+    }
+
+    Logger.log();
+
+    const connectionStatus = await launchAVD(sdkRoot, platform, options.avd as string);
+    if (!connectionStatus) {
+      return false;
+    }
```

Comment:
On doing this, the process gets stuck while the emulator is running. We should instead exit the process and let the emulator run in the background. Then, we can use the `disconnect` command to close the emulator.

Reply:
Could probably use the spawnSync function here... we don't need to kill the already running AVD in this case or wait for the AVD to boot up, so using the `launchAVD` here does not make sense.